api/src/Entity/ScheduleEntry.php

```diff
@@ -136,41 +130,54 @@ public function getCamp(): ?Camp {
     }
 
     /**
-     * The start date and time of the schedule entry. This is a read-only convenience property.
-     *
-     * @return null|DateTime
+     * Start date and time of the schedule entry.
      */
-    #[ApiProperty(writable: false, example: '2022-01-02T00:00:00+00:00', openapiContext: ['format' => 'date'])]
+    #[ApiProperty(example: '2022-01-02T00:00:00+00:00', required: true, openapiContext: ['format' => 'date-time'])]
+    #[Assert\GreaterThanOrEqual(propertyPath: 'period.start')]
     #[Groups(['read'])]
     public function getStart(): ?DateTime {
         try {
             $start = $this->period?->start ? DateTime::createFromInterface($this->period->start) : null;
-            $start?->add(new DateInterval('PT'.$this->periodOffset.'M'));
+            $start?->modify("{$this->startOffset} minutes");
 
             return $start;
         } catch (\Exception $e) {
             return null;
         }
     }
 
+    #[Groups(['write'])]
+    public function setStart(DateTimeInterface $start): void {
+        if (null !== $this->period?->start) {
+            $this->startOffset = DateTimeUtil::differenceInMinutes($this->period->start, $start);
+        }
+    }
+
     /**
-     * The end date and time of the schedule entry. This is a read-only convenience property.
-     *
-     * @return null|DateTime
+     * End date and time of the schedule entry.
      */
-    #[ApiProperty(writable: false, example: '2022-01-02T01:30:00+00:00', openapiContext: ['format' => 'date'])]
+    #[ApiProperty(example: '2022-01-02T01:30:00+00:00', required: true, openapiContext: ['format' => 'date-time'])]
+    #[Assert\GreaterThan(propertyPath: 'start')]
+    #[Assert\LessThanOrEqual(propertyPath: 'period.endOfLastDay')]
     #[Groups(['read'])]
     public function getEnd(): ?DateTime {
         try {
             $end = $this->period?->start ? DateTime::createFromInterface($this->period->start) : null;
-            $end?->add(new DateInterval('PT'.($this->periodOffset + $this->length).'M'));
+            $end?->modify("{$this->endOffset} minutes");
 
             return $end;
         } catch (\Exception $e) {
             return null;
         }
     }
 
+    #[Groups(['write'])]
+    public function setEnd(DateTimeInterface $end): void {
```

Comment:
wenn man hier jetzt zu früh setEnd aufruft, bekommen wir das nicht mit.
Wäre hier eine \RuntimeException angebracht?

Reply:
Dürfte eigentlich nicht passieren. Aber sicher safer, implemented in https://github.com/ecamp/ecamp3/pull/2484/commits/d1fff51c7f200db3ce7e1025ad89731e86cce9fb